corehq/apps/es/case_search.py

```diff
@@ -253,6 +253,18 @@ def reverse_index_case_query(case_ids, identifier=None):
         )
     )
 
+def case_property_exists(case_property_name):
+    """Returns cases where case_property_name is set
+    """
+
+    return queries.nested(
+        CASE_PROPERTIES_PATH,
+        queries.filtered(
+            queries.match_all(),
+            filters.term('{}.key.exact'.format(CASE_PROPERTIES_PATH), case_property_name),
+        )
+    )
```

Comment:
I'd push for pulling the `!= ""` part of the assembled query above into this query.  Conceptually, I think `case_property_exists` also means "case property is not empty", or at least I'm not sure that it's necessary to have both versions of that concept.